packages/tupaia-web-server/src/utils/generateAccessibleCountryFilter.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { AccessPolicy } from '@tupaia/access-policy';
+
+const BES_ADMIN_PERMISSION_GROUP = 'BES Admin';
+
+// Takes an access policy and list of permission groups, generates the list of
+// accessible countries
+// Used to filter results from the entity server
+export function generateAccessibleCountryFilter(
```

Comment:
I originally started doing this work against the entity server but the code wound up getting quite messy. Although I think that was because I was trying to make the whole thing switchable with a parameter. If we're happy having it applied to all users of the entity server I can take another shot at applying the logic in there

Reply:
Yep I reckon we're good to apply it generally across the entity-server if you're happy to give it another shot.

I can't think of a real scenario where we really want to be returning all countries the user has any access to regardless of project. I think we designed that way originally since the idea was that multiple projects share one entity hierarchy. But since we scrapped that a while back I think filtering by project permissions makes a lot of sense 👍